tests/integration/steps/integration_steps.go

```diff
@@ -148,6 +150,59 @@ func WaitForCassDcToBeUpdating(t *testing.T, namespace string) {
 	waitForCassDcToBe(t, namespace, cassdcapi.ProgressUpdating)
 }
 
+// RestartStargate scales the Stargate deployment down to zero and then scales it
+// back up to the prior number of replicas. This function blocks until the
+// deployment is ready.
+func RestartStargate(releaseName, dcName, namespace string) error {
+	key := types.NamespacedName{Namespace: namespace, Name: releaseName + "-" + dcName + "-stargate"}
+	retryInterval := 5 * time.Second
+	scaleDownTimeout := 2 * time.Minute
+	scaleUpTimeout := 5 * time.Minute
+
+	if err := WaitForDeploymentReady(key, retryInterval, scaleDownTimeout); err != nil {
+		return fmt.Errorf("timed out waiting for Stargate to scale down: %s", err)
+	}
+
+	deployment := &appsv1.Deployment{}
+	if err := testClient.Get(context.Background(), key, deployment); err != nil {
+		return fmt.Errorf("failed to get Stargate deployment: %s", err)
+	}
+
+	originalCount := *deployment.Spec.Replicas
+
+	patch := client.MergeFromWithOptions(deployment.DeepCopy(), client.MergeFromWithOptimisticLock{})
+	count := int32(0)
+	deployment.Spec.Replicas = &count
+	if err := testClient.Patch(context.Background(), deployment, patch); err != nil {
+		return fmt.Errorf("failed to scale down Stargate: %s", err)
+	}
+
+
+	patch = client.MergeFromWithOptions(deployment.DeepCopy(), client.MergeFromWithOptimisticLock{})
+	deployment.Spec.Replicas = &originalCount
+	if err := testClient.Patch(context.Background(), deployment, patch); err != nil {
+		return fmt.Errorf("failed to scale up Stargate: %s", err)
+	}
+
+	if err := WaitForDeploymentReady(key, retryInterval, scaleUpTimeout); err != nil {
+		return fmt.Errorf("timed out waiting for Stargate to scale up: %s", err)
+	}
+
+	return nil
+}
+
+// WaitForDeploymentReady Polls the deployment status until the Deployment is
+// ready. Readiness is defined as .Status.Replicas == .Status.ReadyReplicas.
+func WaitForDeploymentReady(key types.NamespacedName, retryInterval, timeout time.Duration) error {
+	return wait.Poll(retryInterval, timeout, func() (bool, error) {
```

Comment:
Maybe it could be good to use `g(t).Eventually()` instead to be consistent with the rest of the test codebase?
Would you see any downside to it?

Reply:
I went ahead and changed it for consistency, but I don't care for the weak typing